nomos-services/cryptarchia-consensus/src/mix/adapters/libp2p.rs

```diff
@@ -0,0 +1,86 @@
+use nomos_mix_service::{
+    backends::libp2p::{
+        Libp2pNetworkBackend, Libp2pNetworkBackendEvent, Libp2pNetworkBackendEventKind,
+        Libp2pNetworkBackendMessage,
+    },
+    NetworkMsg, NetworkService,
+};
+use overwatch_rs::services::{relay::OutboundRelay, ServiceData};
+use tokio::sync::broadcast::{self, error::RecvError};
+use tokio_stream::{wrappers::BroadcastStream, StreamExt};
+
+use crate::mix::{BoxedStream, NetworkAdapter};
+
+const BUFFER_SIZE: usize = 64;
+
+#[derive(Clone)]
+pub struct LibP2pAdapter {
+    network_relay: OutboundRelay<<NetworkService<Libp2pNetworkBackend> as ServiceData>::Message>,
+    mixed_msgs: broadcast::Sender<Vec<u8>>,
+}
+
+#[async_trait::async_trait]
+impl NetworkAdapter for LibP2pAdapter {
+    type Backend = Libp2pNetworkBackend;
+
+    async fn new(
+        network_relay: OutboundRelay<<NetworkService<Self::Backend> as ServiceData>::Message>,
+    ) -> Self {
+        let relay = network_relay.clone();
```

Comment:
No need for clone here I think, `network_relay` is already owned.

Reply:
Fixed in 14b8de118e5eda5e8d1c3a20631e3ea6eff4fd84